File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left =0
        total = 0 
        right =  len(height)-1
        while right >= left:
            res = min(height[left],height[right])*(right-left)
            total = max(res,total)
            if height[left] <= height[right]:
                left+=1
            else:
                right-=1
        return total

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char,int> mp;
        int k = 0;
        int max_seq = 0;
        for(int j=0;j<s.size();j++)
        {
            while(mp[s[j]]==1)
            {
                mp[s[k]]-=1;
                k+=1;
            }
            mp[s[j]]++;
            max_seq= max(max_seq,j-k+1);
        }
        return max_seq;
    }
};
